modules/tinymce/src/core/main/ts/keyboard/PageUpDownKeys.ts

```diff
@@ -0,0 +1,64 @@
+/**
+ * Copyright (c) Tiny Technologies, Inc. All rights reserved.
+ * Licensed under the LGPL or a commercial license.
+ * For LGPL see License.txt in the project root for license information.
+ * For commercial licenses see https://www.tiny.cloud/
+ */
+
+import { Cell } from '@ephox/katamari';
+import Editor from '../api/Editor';
+import { NodeChangeEvent } from '../api/EventTypes';
+import { EditorEvent } from '../api/PublicApi';
+import VK from '../api/util/VK';
+import * as InlineBoundariesNavigation from './InlineBoundariesNavigation';
+import * as MatchKeys from './MatchKeys';
+
+const executeKeyupAction = (editor: Editor, evt: KeyboardEvent) => {
+  MatchKeys.execute([
+    { keyCode: VK.PAGE_UP, action: MatchKeys.action(InlineBoundariesNavigation.moveToLineEndPoint, editor, false) },
```

Comment:
Should this only happen on non macOS platforms? because as mentioned previously macOS causes the page to scroll instead of change the selection. @spocke would be keen to get your thoughts here?

Reply:
The home/end keys seems to move to start/end of document from some stackoverflow post I found. So we need to make sure they are platform specific. Also Mac OS seems to do a real home/end of line if you use fn-arrow left/right and fn-arrow up/down is page up.

I'm unsure if webdriver is automatically remapping these or if these are the same key code and modifier just a way to access them.

So I guess we need to investigate what behavior is the user expecting for real home/end/pg down/pg up keys and the fn-arrow combos are they the same thing, are mac people likely to use, them can we detect all combos if they are different?

Reply:
Sorry, you've probably missed some of the previous discussion. Yes, Fn+Up/Down are page up/down (we get keycode 33/34 in the keydown/keyup events). On a real MacBook pro, using contenteditable, the browser will always scroll instead of moving the selection so I'm thinking we need to keep that behaviour here as well (worth noting it does the same thing in the Pages app).

Reply:
Reading and testing https://developer.apple.com/design/human-interface-guidelines/macos/user-interaction/keyboard/#reserved-and-expected-keyboard-shortcuts, it seems Cmd+Up or Cmd+Down is the closest equivalent to Page Up/Down behaviour in other OS's. As such, we might need to handle that key combination instead of Page Up/Down for macOS.

I should not the above goes to the start/end of the document instead of to the start/end of what's inside the viewport.

Reply:
updated

Reply:
@camiloHimura in that case we need to not run this logic on Mac OS, otherwise it's potentially going to override the page up/down on Mac OS.

Reply:
I think the best option will be to just return immediately in the `setup` function for macOS. Thoughts?

Reply:
I've been testing this implementation on Mac, and the default behavior ( scroll up and down) is not overridden. I don't think it is needed to skip the implementation on Mac explicitly.

Reply:
That's not what would break here, it's that it'd scroll **and** change the selection if it was at the start of an inline boundary. We never want this behaviour to run on Mac, as it doesn't have the same page up/down behaviour and as such this code should never be running on Mac.

Reply:
Ok, in that case, your idea of immediate return on the `setup` is fine.